### PR TITLE
Chore: Propagate context for dashboard guardian

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Require `guardian.New` to take `context.Context` as first argument. Migrates the `GetDashboardAclInfoListQuery` to be dispatched using context.

**Which issue(s) this PR fixes**:
Ref #36734

**Special notes for your reviewer**:
- `GetTeamsByUserQuery` not yet converted to be dispatched with context.Context. PR already to big.

### CODE REVIEW

**Review by sdboyer — APPROVED** [review]
This _seems_ fine to me - the only thing that jumped out as weird was replacing `HTTPServer` with a `dashboards.Store`, though reading just a line further assuaged any concern.

File: pkg/api/acl.go [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+	"github.com/grafana/grafana/pkg/dashboards"
+	"github.com/grafana/grafana/pkg/models"
+)
 
 // updateDashboardACL updates a dashboard's ACL items.
 //
 // Stubbable by tests.
-var updateDashboardACL = func(hs *HTTPServer, dashID int64, items []*models.DashboardAcl) error {
-	return hs.SQLStore.UpdateDashboardACL(dashID, items)
+var updateDashboardACL = func(s dashboards.Store, ctx context.Context, dashID int64, items []*models.DashboardAcl) error {
```
Comment (sakjur) [inline pkg/api/acl.go:13]: Nitpick: Do we want to consistently place context.Context in the first parameter position? (I'd prefer that you'd merge this PR without fixing this if it takes you more than a minute to change this or if you at all disagree)

```suggestion
var updateDashboardACL = func(ctx context.Context, s dashboards.Store, dashID int64, items []*models.DashboardAcl) error {
```

**Review by sakjur — APPROVED** [review]
Not sure what's up with the Markdown formatting, I've seen that elsewhere. It seems like an over-eager commit hook modifying non-modified files?

That aside, this all looks good :+1:

File: pkg/api/acl.go (hunk 1 above)
Comment (sdboyer) [reply pkg/api/acl.go:13]: Oh, agree - context is ALWAYS first parameter. Should've commented on this before approving :)

**Review by sdboyer — DISMISSED** [review]
Change ordering on the parameters

File: pkg/api/dashboard_permission.go [hunk 2]
```diff
@@ -99,7 +100,7 @@ func (hs *HTTPServer) UpdateDashboardPermissions(c *models.ReqContext, apiCmd dt
 		return response.Error(403, "Cannot remove own admin permission for a folder", nil)
 	}
 
-	if err := updateDashboardACL(hs, dashID, items); err != nil {
+	if err := updateDashboardACL(context.Background(), hs.SQLStore, dashID, items); err != nil {
```
Comment (sakjur) [inline pkg/api/dashboard_permission.go:103]:
```suggestion
	if err := updateDashboardACL(c.Req.Context(), hs.SQLStore, dashID, items); err != nil {
```

Why are we not passing the request's context here?

File: pkg/api/folder_permission.go [hunk 3]
```diff
@@ -112,7 +114,7 @@ func (hs *HTTPServer) UpdateFolderPermissions(c *models.ReqContext, apiCmd dtos.
 		return response.Error(403, "Cannot remove own admin permission for a folder", nil)
 	}
 
-	if err := updateDashboardACL(hs, folder.Id, items); err != nil {
+	if err := updateDashboardACL(context.Background(), hs.SQLStore, folder.Id, items); err != nil {
```
Comment (sakjur) [inline pkg/api/folder_permission.go:117]:
```suggestion
	if err := updateDashboardACL(c.Req.Context(), hs.SQLStore, folder.Id, items); err != nil {
```

👆 same question

**Comment by marefr** [comment]
> Not sure what's up with the Markdown formatting, I've seen that elsewhere. It seems like an over-eager commit hook modifying non-modified files

I agree. Seen it in multiple other PRs as well. Must be the pre-commit, potentially related to #37616. Seems like pre-commit/prettier not used by everyone. Regarding the changelog/release notes those are auto-generated by github action so probably some changes needed there to make them "prettier-compliant".

Thanks all. Going to sync up this and the enterprise one with main so that we can get it merged.

**Pull request merged**